pkg/scheduler/internal/queue/events.go

```diff
@@ -70,3 +74,28 @@ const (
 	// NodeConditionChange is the event when node condition is changed.
 	NodeConditionChange = "NodeConditionChange"
 )
+
+var clusterEventReg = map[string]framework.ClusterEvent{
+	AssignedPodAdd:              {Resource: framework.Pod, ActionType: framework.Add},
```

Comment:
Would it be worth replacing these strings with enums so that we can have `[]framework.ClusterEvent` instead? Check with a performance benchmark.

Then we would need to have a `ToString` method for the enum type so that we can still use them in event labels.

Reply:
It's a good point.

As these strings are pre-existing, let me add a TODO here, and refactor in a follow-up if needed.

Reply:
Do we actually need this registry at all? we can construct the event on the fly: instead of passing in `AssignedPodAdd` and then look up the `framework.ClusterEvent`, just pass the cluster event directly (i.e., `{Resource: framework.Pod, ActionType: framework.Add}`), for example here: https://github.com/kubernetes/kubernetes/blob/e1c1fd9edf9bd4d9d0e749b16fb33a54399a1ef8/pkg/scheduler/internal/queue/scheduling_queue.go#L485

And add a ToString on the ClusterEvent type itself for logging and whatnot.

are any of those constants used in more than one place?

Reply:
SGTM.

In this case, we have to change the signature of `SchedulingQueue#MoveAllToActiveOrBackoffQueue(event string)`, which would generate a number of changes. To offload the review work in this PR, mind refactoring it in a follow-up?

Reply:
doing it in a follow up PR sounds good.

Reply:
mind tracking the followups somewhere? perhaps a checklist in a comment in the original issue?

Reply:
+1